crates/swc_ecma_preset_env/src/lib.rs

```diff
@@ -679,9 +678,49 @@ struct CoreJSConfig {
 }
 
 pub struct EnvConfig {
-    config: Config,
     feature_config: Arc<FeatureConfig>,
     core_js_config: CoreJSConfig,
+    mode: Option<Mode>,
+    debug: bool,
+    dynamic_import: bool,
+    loose: bool,
+    core_js: Option<Version>,
+    shipped_proposals: bool,
+}
+
+impl From<&'_ Config> for EnvConfig {
+    fn from(config: &Config) -> Self {
```

Comment:
This should be merged, but well this is just to see if the API looks good enough so I skipped